includes/class-wc-payments.php

```diff
@@ -94,13 +94,19 @@ public static function init() {
 		include_once dirname( __FILE__ ) . '/class-logger.php';
 		include_once dirname( __FILE__ ) . '/class-wc-payment-gateway-wcpay.php';
 		include_once dirname( __FILE__ ) . '/class-wc-payments-token-service.php';
-		include_once WCPAY_ABSPATH . 'includes/exceptions/class-wc-payments-intent-authentication-exception.php';
+		include_once dirname( __FILE__ ) . '/exceptions/class-wc-payments-intent-authentication-exception.php';
 
 		self::$account          = new WC_Payments_Account( self::$api_client );
 		self::$customer_service = new WC_Payments_Customer_Service( self::$api_client );
 		self::$token_service    = new WC_Payments_Token_Service( self::$api_client, self::$customer_service );
 
-		self::$gateway = new WC_Payment_Gateway_WCPay( self::$api_client, self::$account, self::$customer_service, self::$token_service );
+		$gateway_class = 'WC_Payment_Gateway_WCPay';
+		if ( class_exists( 'WC_Subscriptions' ) ) {
+			include_once dirname( __FILE__ ) . '/compat/subscriptions/class-wc-payment-gateway-wcpay-subscriptions-compat.php';
```

Comment:
Please add logic to enforce WC_Subscriptions 3.0.0 or newer cc @RadoslavGeorgiev

Reply:
Fixed in 15ab16df86a01bfe78276e44e6cb8886213f4f32

Reply:
Out of curiosity, which change(s) in 3.0.0 are we depending on?

Reply:
IIRC it had something to do with aligning the PHP versions, but I'm not entirely sure.

cc @allendav @RadoslavGeorgiev to see if they remember

Reply:
We are not depending on any changes in 3.0.0. We just want to limit the scope of testing / support.